src/index.js

```diff
@@ -295,8 +298,18 @@ const snyker = async () => {
   console.log("[SNYKER: STARTING]");
 
   MAX_RETRIES = argv.retries || DEFAULT_RETRIES;
+  PRESERVE_INTEGRITY = argv["preserve-integrity"] || false;
+
+  // We need to determine whether we're using Yarn or NPM
+  // Prioritise "lockfile" flag, then check for yarn.lock, then package-lock.json
+  // If none of these files exist, default to yarn.lock
+  const lockFileName =
+    argv.lockfile ||
+    ["yarn.lock", "package-lock.json"].find((file) =>
+      fs.existsSync(path.join(process.cwd(), file)),
+    ) ||
+    "yarn.lock";
```

Comment:
```suggestion
    );
```
Not sure there's point falling back to `yarn.lock` if just proven it's non-existent...

Reply:
It is important to fallback as it will need to determine if it is doing a `yarn install` or `npm install` as part of the Snyker process. To date, we have defaulted to using `yarn` hence defaulting to that

Reply:
switched it to

`const isYarn = lockFileName === "yarn.lock";`